app/javascript/controllers/file_auth_controller.js

```diff
@@ -264,4 +271,42 @@ export default class extends Controller {
   hideMessagePanel() {
     this.messagePanelTarget.hidden = true
   }
+
+  // To see if item is restricted by location, check the probe service json response
+  handleLocationRestricted(json, accessService) {
+    // The probe auth service is called for each file separately.
+    // If the location restriction target is available, then trigger auth denied message.
+    // The event will lead to the locked icon being displayed for this item
+    // This allows the lock window to show
+    const event = new CustomEvent('auth-denied', { accessService: accessService })
+    window.dispatchEvent(event)
+    // Make the location restricted banner visible
+    this.locationRestrictionTarget.hidden = false
+    // Display the location restriction message based on the authorization response
+    this.locationRestrictionMessageTarget.innerHTML = this.retrieveRestrictedLocationMessage(json)
+
+  }
+
+  // Checks the result of the probe auth request to see if access is restricted to location
+  // This code depends on the text returned by probe service, so changes to the heading
+  // should be reflected here as well.
+  isLocationRestricted(json) {
+    if(json.status == '401' && 'heading' in json && 'en' in json.heading && json.heading.en.length
+      && json.heading.en[0].startsWith('Content is restricted to location'))
+      return true
+    return false
```

Comment:
can we do:
```suggestion
    return json.status == '401' && 'heading' in json && 'en' in json.heading && json.heading.en.length
      && json.heading.en[0].startsWith('Content is restricted to location')
```

Reply:
I'm not sure we need to check `json.heading.en` or `json.heading.en.length` as we know our Stacks response will have these.

Reply:
yes to the first comment.  For the second comment, this is more to ensure if something weird happens on the response end, things don't just stop working.